thorlcr/thorutil/thbuf.cpp

```diff
@@ -1507,16 +1506,15 @@ class CSharedWriteAheadDisk : public CSharedWriteAheadBase
         allocator(rowIf->queryRowAllocator()), deserializer(rowIf->queryRowDeserializer()), serializeMeta(meta->querySerializedDiskMeta())
     {
         assertex(spillName);
-        spillFile.setown(createIFile(spillName));
-        spillFile->setShareMode(IFSHnone);
-        spillFileIO.setown(spillFile->open(IFOcreaterw));
+        Owned<IFile> iFile = createIFile(spillName);
+        iFile->setShareMode(IFSHnone);
+        tempFileIO.setown(iFile->open(IFOcreaterw));
+        tempFileOwner.setown(new CFileOwner(iFile.getClear(), activity->queryTempFileSizeTracker()));
```

Comment:
this now needs to be changed to avoid leaking iFile, e.g. to:
```
tempFileOwner.setown(new CFileOwner(iFile, activity->queryTempFileSizeTracker()));
```